pkg/flare/ip_detector.go

```diff
@@ -0,0 +1,28 @@
+package flare
+
+import (
+	"fmt"
+
+	"github.com/octarinesec/secret-detector/pkg/detectors/helpers"
+	"github.com/octarinesec/secret-detector/pkg/secrets"
+)
+
+const (
+	ipv4 = `\b((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)(?::\d+)?\b`
+	ipv6 = `(([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,7}:|([0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,5}(:[0-9a-fA-F]{1,4}){1,2}|([0-9a-fA-F]{1,4}:){1,4}(:[0-9a-fA-F]{1,4}){1,3}|([0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4}|([0-9a-fA-F]{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}|[0-9a-fA-F]{1,4}:((:[0-9a-fA-F]{1,4}){1,6})|:((:[0-9a-fA-F]{1,4}){1,7}|:)|fe80:(:[0-9a-fA-F]{0,4}){0,4}%[0-9a-zA-Z]{1,}|::(ffff(:0{1,4}){0,1}:){0,1}((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])|([0-9a-fA-F]{1,4}:){1,4}:((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9]))$`
```

Comment:
1. Please take them from some standard package, or just parse IP addresses.
1. What about hostnames?  They are just as secret as IPs.
1. We *must* allow the user to override having these.  Otherwise many configuration errors simply won't show up.

Reply:
1. I couldn't find one. Suggestions are welcome 🙏🏻
2. That's true, but hostnames don't have a similarly strict structure, so I'm not sure we can reliably detect them
3. The assumption is that this will either be for internal use or, if shared externally, will need to be manually reviewed by any sufficiently security-conscious organization. We are making our best effort to assist. This might be good enough for some, but not others. In the documentation for this feature, we'll be completely upfront about this and list exactly what kinds of information is collected, where and how it's collected, and what secrets we can detect/redact from every type of information collected